pkg/controller/resource_sync_task.go

```diff
@@ -155,27 +158,56 @@ func (st *resourceSyncTask) processResource(res *smith_v1.Resource) resourceInfo
 	}
 
 	// Check if resource is ready
-	ready, retriable, err := st.rc.IsReady(resUpdated)
-	if err != nil {
+	if ready, retriable, err := st.rc.IsReady(resUpdated); err != nil {
 		return resourceInfo{
 			actual: resUpdated,
 			status: resourceStatusError{
 				err:              errors.Wrap(err, "readiness check failed"),
 				isRetriableError: retriable,
 			},
 		}
-	}
-	if ready {
+	} else if !ready {
 		return resourceInfo{
 			actual: resUpdated,
-			status: resourceStatusReady{},
+			status: resourceStatusInProgress{},
 		}
-	} else {
+	}
+
+	// Augment with binding output (used for references)
+	bindingSecret, err := st.extractBindingSecret(resUpdated)
+	if err != nil {
 		return resourceInfo{
 			actual: resUpdated,
-			status: resourceStatusInProgress{},
+			status: resourceStatusError{
+				err: err,
+			},
 		}
 	}
+
+	return resourceInfo{
+		actual:               resUpdated,
+		status:               resourceStatusReady{},
+		serviceBindingSecret: bindingSecret,
+	}
+}
+
+func (st *resourceSyncTask) extractBindingSecret(obj *unstructured.Unstructured) (*core_v1.Secret, error) {
+	actual, err := st.scheme.ConvertToVersion(obj, obj.GroupVersionKind().GroupVersion())
```

Comment:
This is calling us to fail CI because the sleeper isn't registered fully with the scheme (or something):

    no kind \"Sleeper\" is registered for version \"crd.atlassian.com/v1\"

I think I should change it to use unstructured OR simply swallow the error (hey, we don't know what type this might be - we might not have registered it with our scheme...), but I'm a little confused here, since I cargo-culted it from the prepareDependencies function in the same file, and it should have the same problem (but not surfaced atm by integration tests). Leaving the breakage here for now for someone who knows more to weigh in.

Reply:
Ok, confirmed that it works if you register the scheme, but I don't think we want to do that.

I think I'll change it to check the error code (and the other one).

Reply:
You are right, I must have introduced a bug in this PR https://github.com/atlassian/smith/pull/228.
No, of course we don't want to register `Sleeper` in our scheme 🤦‍♂️ 
Lets not block this PR on the bug and I'll fix it in a follow up. Sorry for the mess.

Reply:
Also you don't really need to try to convert it to a typed object to check if it is a `ServiceBinding` - just check the GVK. And only then convert. And that will actually help you to not run into the bug :badpokerface: